File: backend/app/schemas/comment.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime


class CommentResponse(BaseModel):
    id: UUID
    comment_id: str
    video_id: UUID
    username: str | None = None
    display_name: str | None = None
    comment_text: str
    likes_count: int
    reply_count: int
    comment_created_at: datetime | None = None
    scraped_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/comment.ts
export interface Comment {
  id: string;
  comment_id: string;
  video_id: string;
  username: string | null;
  display_name: string | null;
  comment_text: string;
  likes_count: number;
  reply_count: number;
  comment_created_at: string | null;
  scraped_at: string;
}

export interface CommentListResponse {
  data: Comment[];
  total: number;
  page: number;
  page_size: number;
  total_pages: number;
}

export interface CommentListParams {
  page?: number;
  page_size?: number;
  search?: string;
  video_id?: string;
  sort_by?: string;
  sort_order?: "asc" | "desc";
}
